<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TraceDialog</class>
 <widget class="QDialog" name="TraceDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>771</width>
    <height>660</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Trace Bitmap</string>
  </property>
  <property name="windowIcon">
   <iconset theme="bitmap-trace">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="layout_main">
     <item>
      <widget class="QWidget" name="widget_controls" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QWidget" name="mode_container" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Mode</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="combo_mode">
             <item>
              <property name="text">
               <string>Transparency</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Closest Color</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Extract Colors</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Pixel</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Image Options</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QStackedWidget" name="stacked_widget">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="currentIndex">
              <number>1</number>
             </property>
             <widget class="QWidget" name="page">
              <layout class="QFormLayout" name="formLayout_2">
               <item row="0" column="0">
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>Output Color</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="color_widgets::ColorSelector" name="color_mono" native="true">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="color" stdset="0">
                  <color>
                   <red>0</red>
                   <green>0</green>
                   <blue>0</blue>
                  </color>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>Invert</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QCheckBox" name="check_inverted">
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>Threshold</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QSpinBox" name="spin_alpha_threshold">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="maximum">
                  <number>255</number>
                 </property>
                 <property name="value">
                  <number>128</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="page_2">
              <layout class="QVBoxLayout" name="verticalLayout_5">
               <item>
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>Colors</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QListWidget" name="list_colors">
                 <property name="dragEnabled">
                  <bool>true</bool>
                 </property>
                 <property name="dragDropMode">
                  <enum>QAbstractItemView::InternalMove</enum>
                 </property>
                 <property name="defaultDropAction">
                  <enum>Qt::MoveAction</enum>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_4">
                 <item>
                  <widget class="QToolButton" name="toolButton">
                   <property name="toolTip">
                    <string>Add Color</string>
                   </property>
                   <property name="text">
                    <string>Add</string>
                   </property>
                   <property name="icon">
                    <iconset theme="list-add">
                     <normaloff>.</normaloff>.</iconset>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="horizontalSpacer">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="QToolButton" name="toolButton_2">
                   <property name="toolTip">
                    <string>Remove Color</string>
                   </property>
                   <property name="text">
                    <string>Remove</string>
                   </property>
                   <property name="icon">
                    <iconset theme="list-remove">
                     <normaloff>.</normaloff>.</iconset>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QGridLayout" name="gridLayout">
                 <item row="3" column="2">
                  <widget class="QSpinBox" name="spin_color_count">
                   <property name="maximum">
                    <number>32</number>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="2">
                  <widget class="QDoubleSpinBox" name="spin_outline">
                   <property name="singleStep">
                    <double>0.500000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="0" colspan="2">
                  <widget class="QLabel" name="label_tolerance">
                   <property name="text">
                    <string>Tolerance</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0" colspan="2">
                  <widget class="QLabel" name="label_9">
                   <property name="text">
                    <string>Outline</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="2">
                  <widget class="QSpinBox" name="spin_tolerance">
                   <property name="minimum">
                    <number>0</number>
                   </property>
                   <property name="maximum">
                    <number>512</number>
                   </property>
                   <property name="singleStep">
                    <number>10</number>
                   </property>
                   <property name="value">
                    <number>10</number>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QPushButton" name="button_detect_colors">
                   <property name="text">
                    <string>Detect Colors</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="0">
                  <widget class="QToolButton" name="button_quantize_options">
                   <property name="toolTip">
                    <string>Configure Color Detection</string>
                   </property>
                   <property name="text">
                    <string>Configure...</string>
                   </property>
                   <property name="icon">
                    <iconset theme="settings-configure">
                     <normaloff>.</normaloff>.</iconset>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="page_3"/>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="group_potrace">
          <property name="title">
           <string>Curve Options</string>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Smoothness</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QDoubleSpinBox" name="spin_smoothness">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="suffix">
              <string>%</string>
             </property>
             <property name="decimals">
              <number>0</number>
             </property>
             <property name="maximum">
              <double>100.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>5.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="spin_min_area">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Minimum Area</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="group_preview">
       <property name="title">
        <string>Preview</string>
       </property>
       <layout class="QVBoxLayout" name="group_preview_layout">
        <item>
         <widget class="QGraphicsView" name="preview">
          <property name="renderHints">
           <set>QPainter::Antialiasing|QPainter::TextAntialiasing</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="preview_slider">
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progress_bar">
          <property name="value">
           <number>24</number>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QPushButton" name="button_update_preview">
            <property name="text">
             <string>Update</string>
            </property>
            <property name="icon">
             <iconset theme="document-preview">
              <normaloff>.</normaloff>.</iconset>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_preview_zoom">
            <property name="text">
             <string>Zoom</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="spin_preview_zoom">
            <property name="suffix">
             <string>%</string>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
            <property name="stepType">
             <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
            </property>
            <property name="value">
             <double>100.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="layout_buttons">
     <item>
      <widget class="QPushButton" name="button_help">
       <property name="text">
        <string>Help...</string>
       </property>
       <property name="icon">
        <iconset theme="help-contents">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="button_defaults">
       <property name="text">
        <string>Revert to Defaults</string>
       </property>
       <property name="icon">
        <iconset theme="kt-restore-defaults">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>color_widgets::ColorPreview</class>
   <extends>QWidget</extends>
   <header>QtColorWidgets/color_preview.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>color_widgets::ColorSelector</class>
   <extends>color_widgets::ColorPreview</extends>
   <header>QtColorWidgets/color_selector.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>button_update_preview</sender>
   <signal>clicked()</signal>
   <receiver>TraceDialog</receiver>
   <slot>update_preview()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>396</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>534</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton</sender>
   <signal>clicked()</signal>
   <receiver>TraceDialog</receiver>
   <slot>add_color()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>345</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>625</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton_2</sender>
   <signal>clicked()</signal>
   <receiver>TraceDialog</receiver>
   <slot>remove_color()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>281</x>
     <y>345</y>
    </hint>
    <hint type="destinationlabel">
     <x>339</x>
     <y>625</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_detect_colors</sender>
   <signal>clicked()</signal>
   <receiver>TraceDialog</receiver>
   <slot>auto_colors()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>445</y>
    </hint>
    <hint type="destinationlabel">
     <x>159</x>
     <y>625</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_preview_zoom</sender>
   <signal>valueChanged(double)</signal>
   <receiver>TraceDialog</receiver>
   <slot>zoom_preview(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>754</x>
     <y>600</y>
    </hint>
    <hint type="destinationlabel">
     <x>921</x>
     <y>636</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_help</sender>
   <signal>clicked()</signal>
   <receiver>TraceDialog</receiver>
   <slot>show_help()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>634</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>714</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>TraceDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>594</x>
     <y>643</y>
    </hint>
    <hint type="destinationlabel">
     <x>616</x>
     <y>779</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>TraceDialog</receiver>
   <slot>apply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>513</x>
     <y>633</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>798</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>preview_slider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>TraceDialog</receiver>
   <slot>preview_slide(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>671</x>
     <y>513</y>
    </hint>
    <hint type="destinationlabel">
     <x>1068</x>
     <y>536</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_defaults</sender>
   <signal>clicked()</signal>
   <receiver>TraceDialog</receiver>
   <slot>reset_settings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>116</x>
     <y>643</y>
    </hint>
    <hint type="destinationlabel">
     <x>156</x>
     <y>763</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_quantize_options</sender>
   <signal>clicked()</signal>
   <receiver>TraceDialog</receiver>
   <slot>color_options()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>38</x>
     <y>451</y>
    </hint>
    <hint type="destinationlabel">
     <x>108</x>
     <y>815</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>update_preview()</slot>
  <slot>apply()</slot>
  <slot>change_mode(int)</slot>
  <slot>add_color()</slot>
  <slot>remove_color()</slot>
  <slot>auto_colors()</slot>
  <slot>zoom_preview(double)</slot>
  <slot>show_help()</slot>
  <slot>preview_slide(int)</slot>
  <slot>reset_settings()</slot>
  <slot>color_options()</slot>
 </slots>
</ui>
